csrc/ir/nodes.cpp

```diff
@@ -5726,61 +5726,100 @@ std::vector<PolymorphicValue> GroupedMmaOp::evaluate(
+          "GroupedMmaOp expects tensor input at position 4 but got ",
+          inputs[4].type().name());
+
+      auto scale1 = inputs[scale1Offset()].as<at::Tensor>();
+      auto scale2 = inputs[scale2Offset()].as<at::Tensor>();
+      // Note: at::_scaled_grouped_mm requires k dimension to be the fastest on
+      // both input matrices.
+      auto mat1_k_last = mat1.contiguous();
+      auto mat2_k_last = mat2.transpose(-1, -2).contiguous().transpose(-1, -2);
+
+      // at::_scaled_grouped_mm limitation
+      NVF_CHECK(
+          scale1.size(-1) == 1 && scale2.size(-2) == 1,
+          "Scale1 and scale2 must have size 1 at the k dimension. Got ",
+          scale1.sizes(),
+          " and ",
+          scale2.sizes());
+      // scale factor handling
+      // see NOTE -- [ Grouped Matrix Multiplication semantics ]
+      if (TensorDomain::noReductions(out()->getLogicalDomain()).size() == 3) {
+        // case 1, aten API expects collapsed 1D scale with group dimension on
+        // the slower side.
+        scale1 = scale1.reshape(-1);
+        scale2 = scale2.reshape(-1);
+      } else {
+        // case 2 and 3, aten doesn't allow broadcast on k dimension. squeeze k
+        // out.
+        scale1 = scale1.squeeze(-1);
+        scale2 = scale2.squeeze(-2);
+      }
+      // undefined alpha, bias is not supported by aten API
+      NVF_ERROR(!alpha.defined(), "alpha is not supported yet");
+      NVF_ERROR(!beta.defined(), "beta is not supported yet");
+      NVF_ERROR(!bias.defined(), "bias is not supported yet");
+      // NOTE: at::_scaled_grouped_mm only supports bfloat16 as output at this
+      // moment, otherwise we should have requested the output dtype directly
+      // instead of casting the output afterwards.
+      return at::_scaled_grouped_mm(
+          mat1_k_last,
+          mat2_k_last,
+          scale1,
+          scale2,
+          offsets,
+          /*bias=*/std::nullopt,
+          /*alpha=*/std::nullopt,
+          at::ScalarType::BFloat16);
+#else
+      NVF_THROW("at::_scaled_grouped_mm does not exist prior to PyTorch 2.8.");
+#endif
     }
-    // undefined alpha, bias is not supported by aten API
-    NVF_ERROR(!alpha.defined(), "alpha is not supported yet");
-    NVF_ERROR(!beta.defined(), "beta is not supported yet");
-    NVF_ERROR(!bias.defined(), "bias is not supported yet");
-    // NOTE: at::_scaled_grouped_mm only supports bfloat16 as output at this
-    // moment, otherwise we should have requested the output dtype directly
-    // instead of casting the output afterwards.
-    result = at::_scaled_grouped_mm(
-        mat1_k_last,
-        mat2_k_last,
-        scale1,
-        scale2,
-        offsets,
-        /*bias=*/std::nullopt,
-        /*alpha=*/std::nullopt,
-        at::ScalarType::BFloat16);
-  } else {
+
+#if NVFUSER_CUTLASS_KERNEL_ENABLED
+    auto supported_by_cutlass_kernel = [&]() -> bool {
+      return at::cuda::getCurrentDeviceProperties()->major >= 10 &&
```

Comment:
Add `at::cuda::getCurrentDeviceProperties()->major < 11` to avoid issues with devices like 5090